event-log/src/main/scala/io/renku/eventlog/subscriptions/SubscriptionCategoryRegistry.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020 Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.renku.eventlog.subscriptions
+
+import cats.effect.{ContextShift, Effect, IO, Timer}
+import cats.syntax.all._
+import cats._
+import ch.datascience.db.{DbTransactor, SqlQuery}
+import ch.datascience.graph.model.projects
+import ch.datascience.metrics.{LabeledGauge, LabeledHistogram}
+import io.chrisdavenport.log4cats.Logger
+import io.circe.Json
+import io.renku.eventlog.EventLogDB
+
+import scala.concurrent.ExecutionContext
+
+trait SubscriptionCategoryRegistry[Interpretation[_]] {
+
+  def run(): Interpretation[Unit]
+
+  def register(subscriptionRequest: Json): Interpretation[Either[RequestError, Unit]]
+}
+
+private[subscriptions] class SubscriptionCategoryRegistryImpl[Interpretation[_]: Effect: Applicative](
+    categories: Set[SubscriptionCategory[Interpretation]]
+) extends SubscriptionCategoryRegistry[Interpretation] {
+  override def run(): Interpretation[Unit] = categories.toList.traverse_(_.run())
+
+  override def register(subscriptionRequest: Json): Interpretation[Either[RequestError, Unit]] =
+    if (categories.isEmpty) {
+      Either.left[RequestError, Unit](NoCategoriesAvailable).pure[Interpretation]
+    } else {
+      categories.toList
+        .map(_.register(subscriptionRequest).map {
+          case Some(_) => Some(())
```

Comment:
Not totally convinced about this part there may be a better way of doing this. Not sure why but sequence does not understand the abstract type SubscriptionCategory#PayloadType. There could be an implicit missing